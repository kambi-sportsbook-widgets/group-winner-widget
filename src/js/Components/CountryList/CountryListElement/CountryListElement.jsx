import { OutcomeButtonUI } from 'kambi-widget-components';
import React from 'react';
import styles from './CountryListElement.scss';

const CountryListElement = ({ country, value, flagUrl }) => {
   return (
      <tr className={styles.general}>
         <td className={styles.flag}>
            <img
               role='presentation'
               className='l-mr-12'
               width='24'
               height='24'
               src={flagUrl} />
         </td>
         <td className={styles.country}>{ country }</td>
         <td className={styles.button}><OutcomeButtonUI odds={value} /></td>
      </tr>)
};

CountryListElement.propTypes = {
   country: React.PropTypes.node,
   flagUrl: React.PropTypes.string,
   value: React.PropTypes.string
};

export default CountryListElement;
