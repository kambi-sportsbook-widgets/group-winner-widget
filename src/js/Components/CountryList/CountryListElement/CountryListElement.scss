:local(.row) {
   height: 44px;
   cursor: pointer;

   :local(.flag) {
      line-height: 1;
      padding-right: 12px;
      width: 36px;

      img {
         width: 24px;
         height: 24px;
      }
   }

   :local(.country) {
      font-size: 14px;
   }

   :local(.button) {
      width: 80px;
      padding-left: 3px; // makes the space between outcome buttons 3px

      :local(.wrapper) {
         display: flex;
      }
   }
}
