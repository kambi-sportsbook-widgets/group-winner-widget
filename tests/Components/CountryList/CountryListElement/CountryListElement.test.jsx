/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import CountryListElement from '../../../../src/js/Components/CountryList/CountryListElement/CountryListElement';

let renderer;

describe('CountryListElement.render()', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly', () => {
      expect(renderer.render(
         <CountryListElement
            country="Poland!"
            flagUrl="url/to/flag/on/poland.jpg"
            outcome={{id: 100}}
            onClick={() => {}}
         />
      )).toMatchSnapshot();
   });

});

describe('CountryListElement.onClick()', () => {

   it('handles clicks correctly', () => {
      const mockOnClick = jest.fn();

      const wrapper = shallow(
         <CountryListElement
            country="Poland!"
            flagUrl="url/to/flag/on/poland.jpg"
            outcome={{id: 100}}
            onClick={mockOnClick}
         />
      );

      expect(mockOnClick).not.toHaveBeenCalled();

      wrapper.find('.flag').simulate('click');
      expect(mockOnClick).toHaveBeenCalledTimes(1);
      mockOnClick.mockClear();

      wrapper.find('.country').simulate('click');
      expect(mockOnClick).toHaveBeenCalledTimes(1);
      mockOnClick.mockClear();

      wrapper.find('.button').simulate('click');
      expect(mockOnClick).not.toHaveBeenCalled();
   });

});
