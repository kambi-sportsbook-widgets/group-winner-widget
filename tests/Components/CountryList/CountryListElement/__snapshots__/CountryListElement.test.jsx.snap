// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryListElement.render() renders correctly 1`] = `
<li
  className="row"
>
  <div
    className="flag"
    onClick={[Function]}
  >
    <img
      role="presentation"
      src="url/to/flag/on/poland.jpg"
    />
  </div>
  <div
    className="country"
    onClick={[Function]}
  >
    Poland!
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      event={Object {}}
      label={false}
      outcome={
        Object {
          "id": 100,
        }
      }
      outlineStyle={false}
      updateOdds={false}
    />
  </div>
</li>
`;
